### PR TITLE
✨Source FreshServices : Migrate Python CDK to Low-code CDK

### LINKED ISSUE #28992: freshservice
Please use the [connector builder](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview/?_gl=1*13hxonw*_gcl_au*MTI0NTMzNjM3OS4xNjkwODM4MzE0*_ga*MTg2NTc3OTc0Ni4xNjUxNTk0NTQw*_ga_HDBMVFQGBH*MTY5MTAwNDEzMy4yNzIuMC4xNjkxMDA0MTMzLjAuMC4w&_ga=2.197654767.1978878.1690818922-1865779746.1651594540) for this.

### PR DESCRIPTION
**Pull request by avirajsingh7**
## What
Migrating Source freshservices to Low-Code CDK
Closes #28992 

## How
Developed using (Configuration Based Source) low-code CDK

## Recommended reading order

1. spec.yaml
2. manifest.yaml
3. schemas/*

## 🚨 User Impact 🚨

*Are there any breaking changes? What is the end result perceived by the user?*
Yes, there are breaking changes due to Changes to Stream Schema
At many places previous data type were wrong and added some new properties in schema as per response from API

 ##  Test
<details>
<summary>Acceptance Test</summary>
Screenshot 2023-08-06 at 3 59 04 PM
</details>

<details>
<summary>Migrate Connector : Source FreshServices</summary>

- [x] **Community Member? Grant Edit Access to Maintainers [instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests)**
- [x] **Secrets in the Connector's Spec**
- [x] **Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run ./gradlew :airbyte-integrations:connectors:<name>:integrationTest.**
- [x] **Code Reviews Completed**
  - [x] **Documentation Updated**
  - [x] **Connector's README.md**
  - [x] **Connector's bootstrap.md**
  - [x] **docs/integrations/<source or destination>/<name>.md (including changelog. See changelog example)**
  - [x] **docs/integrations/README.md**
  - [x] **airbyte-integrations/builds.md**
- [x] **PR Name follows PR Naming Conventions**
</details>

### CODE REVIEW

File: docs/integrations/sources/freshservice.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ Please read [How to find your API key](https://api.freshservice.com/#authenticat
 
 | Version | Date | Pull Request | Subject |
 | :--- | :--- | :--- | :--- |
+| 2.0.0 | 2023-08-06 | [29126](https://github.com/airbytehq/airbyte/pull/29126) | Migrated to Low-Code CDK |
```
Comment (archangelic) [inline docs/integrations/sources/freshservice.md:57]: This will be a minor version update to `1.2.0`.

**Review by archangelic — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! This overall looks great! I have a few small changes for you!

File: airbyte-integrations/connectors/source-freshservice/metadata.yaml [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
   githubIssueLabel: source-freshservice
   icon: freshservice.svg
   license: MIT
   name: Freshservice
-  registries:
-    cloud:
-      enabled: true
-    oss:
-      enabled: true
+  releaseDate: TODO
```
[the remark this answers is not in the document]
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-freshservice/metadata.yaml]: set to intial release date

File: airbyte-integrations/connectors/source-freshservice/acceptance-test-config.yml [hunk 3]
```diff
@@ -1,37 +1,33 @@
...
-          "satisfaction_survey_responses",
-        ]
-  incremental:
-    - config_path: "secrets/config.json"
-      configured_catalog_path: "integration_tests/configured_catalog.json"
-      future_state_path: "integration_tests/abnormal_state.json"
+    tests:
+      - config_path: "secrets/config.json"
+        configured_catalog_path: "integration_tests/configured_catalog.json"
+        empty_streams: ["name":"satisfaction_survey_responses","name":"releases"]
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-freshservice/acceptance-test-config.yml:23]:
```suggestion
        empty_streams:
          - name: satisfaction_survey_responses
            bypass_reason: Test account does not have permissions
          - name: releases
            bypass_reason: Test account does not have permissions
          - name: assets
            bypass_reason: Test account does not have permissions
          - name: problems
            bypass_reason: Test account does not have permissions
          - name: software
            bypass_reason: Test account does not have permissions
          - name: products
            bypass_reason: Test account does not have permissions
          - name: changes
            bypass_reason: Test account does not have permissions
          - name: vendors
            bypass_reason: Test account does not have permissions
          - name: purchase_orders
            bypass_reason: Test account does not have permissions
```
Add empty streams

File: airbyte-integrations/connectors/source-freshservice/acceptance-test-config.yml [hunk 4]
```diff
@@ -1,37 +1,33 @@
...
+  incremental: 
+   tests:
+     - config_path: "secrets/config.json"
+       configured_catalog_path: "integration_tests/configured_catalog.json"
+       future_state:
+         future_state_path: "integration_tests/abnormal_state.json"
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-freshservice/acceptance-test-config.yml:29]:
```suggestion
  incremental: 
    tests:
      - config_path: "secrets/config.json"
        configured_catalog_path: "integration_tests/configured_catalog.json"
        future_state:
          future_state_path: "integration_tests/abnormal_state.json"
```
Fix indentation

File: airbyte-integrations/connectors/source-freshservice/source_freshservice/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      type: DpathExtractor
+      field_path: ["{{ parameters.path_extractor }}"]
+  requester:
+    type: HttpRequester
+    url_base: "https://{{config['domain_name']}}/api/v2/"
+    http_method: "GET"
+    authenticator:
+      type: "BasicHttpAuthenticator"
+      username: "{{config['api_key']}}"
+      password: ""
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-freshservice/source_freshservice/manifest.yaml:16]:
```suggestion
      password: ""
    error_handler:
      response_filters:
        - http_codes: [403]
          action: IGNORE
```
add error handling for streams with no permissions.

**Review by archangelic — CHANGES_REQUESTED** [review]
I have added some changes to get acceptance tests working!

**Comment by avirajsingh7** [comment]
@archangelic anything else

**Review by archangelic — APPROVED** [review]

**Comment by archangelic** [comment]
Everything looks good! Going to merge this!

**Pull request merged**